packages/tc-ui/src/primitives/relationship/lib/view-relationship.jsx

```diff
@@ -1,25 +1,76 @@
+/* eslint-disable no-sequences, no-return-assign */
 const React = require('react');
 const sortBy = require('lodash.sortby');
 const { getType } = require('@financial-times/tc-schema-sdk');
-const { LinkToRecord } = require('../../../lib/components/structure');
+const {
+	LinkToRecord,
+	LabelledPrimitive,
+} = require('../../../lib/components/structure');
 
 let RelationshipAnnotator;
 
-const OneRelationship = ({ type, value = {}, id }) =>
-	type && value[type] ? (
+const OneRelationship = props => {
+	const {
+		assignComponent,
```

Comment:
you mention in a comment before that you'd like to find a better way to pass assignComponent around. I agree that this is a little ugly, but can't think of a better way either. 

@ericanand do you think React hooks might help us be able to pass a reference to a utility function around?

Reply:
I probably need more context about this, but if you're finding your passing something around alot, maybe [contexts](https://reactjs.org/docs/context.html) is what you want? (I feel I missed out on a joke here with me lacking context about this but oh well)

Happy to chat about this.

Reply:
Will have a look into that